web-services-app/src/test/java/edu/unc/lib/boxc/web/services/processing/SingleUseKeyServiceTest.java

```diff
@@ -0,0 +1,129 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * @author snluong
+ */
+public class SingleUseKeyServiceTest {
+    @TempDir
+    public Path tmpFolder;
+    private Path csvPath;
+
+    private static final String UUID_1 = "f277bb38-272c-471c-a28a-9887a1328a1f";
+    private static final String UUID_2 = "ba70a1ee-fa7c-437f-a979-cc8b16599652";
+    private static final String UUID_3 = "83c2d7f8-2e6b-4f0b-ab7e-7397969c0682";
+    private static final String UUID_TEST = "0e33ad0b-7a16-4bfa-b833-6126c262d889";
+    private String[] ids = {UUID_1, UUID_2, UUID_3};
+    private SingleUseKeyService singleUseKeyService;
+
+    @BeforeEach
+    public void setup() throws IOException {
+        csvPath = tmpFolder.resolve("singleUseKey");
+        singleUseKeyService = new SingleUseKeyService();
+        singleUseKeyService.setCsvPath(csvPath);
+    }
+
+    @Test
+    public void testGenerate() throws IOException {
+        generateDefaultCsv(null);
+        var oldRecords = parseCsv(CSV_HEADERS, csvPath);
+        assertDoesNotContainValue(oldRecords, SingleUseKeyService.ID, UUID_TEST);
+
+        var expirationTime = getExpirationInMilliseconds();
+        var key = singleUseKeyService.generate(UUID_TEST, expirationTime);
+        var newRecords = parseCsv(CSV_HEADERS, csvPath);
+        assertContainsAccessKeyPair(newRecords, UUID_TEST, key);
+    }
+
+    @Test
+    public void testKeyIsValid() throws IOException {
+        var key = SingleUseKeyService.getKey();
+        generateDefaultCsv(key);
+        var currentMilliseconds = System.currentTimeMillis();
+        assertTrue(singleUseKeyService.keyIsValid(UUID_TEST, key, currentMilliseconds));
+    }
+
+    @Test
+    public void testKeyIsNotValid() throws IOException {
+        var key = SingleUseKeyService.getKey();
+        generateDefaultCsv(null);
+        var currentMilliseconds = System.currentTimeMillis();
+        assertFalse(singleUseKeyService.keyIsValid(UUID_TEST, key, currentMilliseconds));
+    }
+
+    @Test
+    public void testKeyIsNotValidWrongUUID() throws IOException {
+        var key = SingleUseKeyService.getKey();
+        generateDefaultCsv(key);
+        var currentMilliseconds = System.currentTimeMillis();
+        assertFalse(singleUseKeyService.keyIsValid(
+                "8e0040b2-9951-48a3-9d65-780ae7106951", key, currentMilliseconds));
+    }
+
+    @Test
+    public void testKeyIsNotValidCurrentTimeIsMoreThan24hLater() throws IOException {
+        var key = SingleUseKeyService.getKey();
+        generateDefaultCsv(key);
+        var currentMilliseconds = System.currentTimeMillis() + (3 * DAY_MILLISECONDS);
+        assertFalse(singleUseKeyService.keyIsValid(UUID_TEST, key, currentMilliseconds));
+    }
+
+    @Test
+    public void testInvalidate() throws IOException {
```

Comment:
It would be good to add a test for when the key does not exist in the csv file to define what that behavior should be. Having nothing happen (current behavior?) is fine, since I think both of the cases we're going to use invalidate will have already checked that the key existed right before invalidating it, but its still useful to define the expected outcome in a test. Also, we could skip overwriting to the file if nothing was removed.